sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionLoadBalancerImpl.java

```diff
@@ -111,30 +111,30 @@ private Mono<Void> run(CancellationToken cancellationToken) {
                         if (cancellationToken.isCancellationRequested()) return Mono.empty();
                         return this.partitionController.addOrUpdateLease(lease);
                     })
-                    .then(Mono.just(this)
-                        .flatMap(value -> {
-                            if (cancellationToken.isCancellationRequested()) {
-                                return Mono.empty();
-                            }
-
-                            Instant stopTimer = Instant.now().plus(this.leaseAcquireInterval);
-                            return Mono.just(value)
-                                .delayElement(Duration.ofMillis(100), CosmosSchedulers.COSMOS_PARALLEL)
-                                .repeat( () -> {
-                                    Instant currentTime = Instant.now();
-                                    return !cancellationToken.isCancellationRequested() && currentTime.isBefore(stopTimer);
-                                }).last();
-                        })
-                    );
+                    .then();
```

Comment:
Do we need this `then()` now, given we have a new `then()` down below?

Reply:
You'll still need the then --> The then operator strips the "final" completion signal of the flux and exposes it as a Mono. otherwise the onErrorResume would happen for every signal of the flux - not just once. At least that is my understanding - @xinlian12  can probably better explain or correct me ;-)

Reply:
and it will complain the return type mismatch >_<

Reply:
Something is not write here... we were only supposed to add the missing throttling in the onErrorResume such that it prevents the load balancer worker thread to making too many requests towards the lease container...

Reply:
I combined both cases. Earlier we only do the delay on success case, now we will also need to do on error case. Since they are using the same logic, so I combined into one